### PR TITLE
[OC-1663] Removing Ribbon and updating Spring Boot to 2.4.6

### PR DESCRIPTION
**Pull request by AlexGuironnetRTE**
Quick recap of the changes in this PR:

- Upgrading Spring Boot beyond 2.4.0-M1 meant we needed to remove Ribbon as it is no longer supported
- Ribbon was also pulling other "utility" dependencies that we made use of in other parts of the code, so it caused errors when I removed it.
  - In the case of JsonSmart, I added it as a direct dependency as we were using it in several places
  - org.apache.commons.lang.StringUtils was used for a single "isEmpty" check on a string so I removed it
- Moved guava dependency definition to main build file like the others
- Updating Spring seems to have updated the Kafka version as well, and one of the test regarding configuration was no longer passing. I fixed it in the spirit of what was already done in the test, but maybe @JeroenGommans you could have a look ?
- The spring boot update meant spring cloud bus was updated as well with a breaking change to the ServiceMatcher class (became an interface) that was causing the unit tests to fail. Mocking the behaviour from the getBusId method solved that (as we're not really testing this in the unit tests, a dummy service id is ok I think).

Karate Tests and Cypress Tests are passing.

I didn't go to Spring Boot 2.5.0 in the end because even though it brought no new issues with the build, there seems to be a breaking change with the associated version of the spring boot gradle plugin that caused the jars not to work with docker, so I left some fun for another time ;-)

In release notes: 
OC-1663 : Removing Ribbon and updating Spring Boot to 2.4.6
Please see the migration guide regarding the configuration of a new mandatory deployment property.
//TODO Add link

It's a breaking change in a way because it's a new property and it's mandatory, but I don't know if it's worth making a migration guide just for this change?

Signed-off-by: Alexandra Guironnet <alexandra.guironnet@rte-france.com>

### CODE REVIEW

**Comment by AlexGuironnetRTE** [comment]
As discussed: 
- Renamed property and moved it to common property file : Done & tested in docker mode

TODO:
- Test dev mode
- Migration guide & release notes
- Error message if property not set?

**Comment by AlexGuironnetRTE** [comment]
Test (docker-mode) if property not set:

Looking at the logs of business services, they all have "Application run failed" tracing back to: 

`org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'webSecurityConfiguration': Unsatisfied dependency expressed through field 'webSecurityChecks'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.opfab.springtools.configuration.oauth.OAuth2GenericConfiguration': Unsatisfied dependency expressed through field 'userServiceCache'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.opfab.springtools.configuration.oauth.UserServiceCache': Unsatisfied dependency expressed through field 'proxy'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.opfab.springtools.configuration.oauth.UserServiceProxy': Unexpected exception during bean creation; nested exception is java.lang.IllegalArgumentException: Could not resolve placeholder 'operatorfabric.servicesUrls.users' in value "${operatorfabric.servicesUrls.users}"
`
So it's not pretty, but it does point to the missing property quite clearly.

What's a bit puzzling though is that if I do a docker-compose ps, it shows that businessconfig and cards-publication exited with 1 as expected since the app couldn't start but cards-consult is still running...

image

File: src/docs/asciidoc/resources/migration_guide_to_2.6.adoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+[source,yaml]
+----
+operatorfabric:
+  servicesUrls:
+    users: "http://localhost:2103"
+----
+
+According to the Feign documentation, the property should contain "an absolute URL or resolvable hostname (the protocol is optional)".
+
+IMPORTANT: This property is mandatory, if it is absent the application won't start.
```
[the remark this answers is not in the document]
Comment (AlexGuironnetRTE) [reply src/docs/asciidoc/resources/migration_guide_to_2.6.adoc:33]: True! Done.

File: services/core/cards-publication/src/main/resources/application.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+spring:
+  application:
+    name: cards-publication
```
[the remark this answers is not in the document]
Comment (AlexGuironnetRTE) [reply services/core/cards-publication/src/main/resources/application.yml:3]: I don't think so as my understanding is that the property files under src/main/resources just get bundled in the jar, so I couldn't see any situation where people would build the jars but not from a clone of our repo (in which case they will get the card-pub application.yml automatically). What do you think?
I can add it to the migration doc anyway but I didn't want to confuse people.

[the remark this answers is not in the document]
Comment (freddidierRTE) [reply services/core/cards-publication/src/main/resources/application.yml:3]: ok

**Pull request merged**